### PR TITLE
[src] Make sample rate positional

### PR DESCRIPTION
**Pull request by mpariente**
As discussed in #427, we'll remove default sample rate argument. 

We start with a deprecation warning when it is not passed, and will remove it in a later release.

### CODE REVIEW

**Comment by jonashaag** [comment]
Positional and having a default are orthogonal concepts. I think the wording should be: “will be required argument in the future.”

Also, is it safe to remove the backwards compat SR table lookups now? (I am not up to date on that front)

**Comment by mpariente** [comment]
> Positional and having a default are orthogonal concepts. I think the wording should be: “will be required argument in the future.”

Thanks.
I was thinking about the wording and I was sure I messed it up :wink: 

> Also, is it safe to remove the backwards compat SR table lookups now? (I am not up to date on that front)

No, the only model that has 16k FS is DeMask, which is not on HF's hub yet (@popcornell ? ^^)

File: asteroid/models/base_models.py [hunk 1]
```diff
@@ -37,8 +37,16 @@ class BaseModel(torch.nn.Module):
             If None, no checks will be performed.
     """
 
-    def __init__(self, sample_rate: float = 8000.0, n_channels: Optional[int] = 1):
+    def __init__(self, sample_rate: float = None, n_channels: Optional[int] = 1):
         super().__init__()
+        if sample_rate is None:
+            sample_rate = 8000.0
+            warnings.warn(
+                "The argument `sample_rate` of `BaseModel` will be made positional. "
```
Comment (mpariente) [inline asteroid/models/base_models.py:45]:
```suggestion
                "The argument `sample_rate` of `BaseModel` will be required in the future. "
```

File: asteroid/models/base_models.py [hunk 2]
```diff
@@ -37,8 +37,16 @@ class BaseModel(torch.nn.Module):
...
+    def __init__(self, sample_rate: float = None, n_channels: Optional[int] = 1):
         super().__init__()
+        if sample_rate is None:
+            sample_rate = 8000.0
+            warnings.warn(
+                "The argument `sample_rate` of `BaseModel` will be required in the future. "
+                "It is no longer a keyword argument. This will raise an error in future release. "
+                "Defaults to 8000.0",
+                VisibleDeprecationWarning,
+            )
```
Comment (mpariente) [inline asteroid/models/base_models.py:49]: How does it look now @jonashaag ?

Comment (jonashaag) [reply asteroid/models/base_models.py:49]: :+1:

**Pull request merged**